packages/migrate/src/SchemaEngineWasm.ts

```diff
@@ -67,14 +67,18 @@ export class SchemaEngineWasm implements SchemaEngine {
 
     // Note: `datamodels` must be either `undefined` or a *non-empty* `LoadedFile[]`.
     const datamodels = schemaContext?.schemaFiles
-    const engine = await wasmSchemaEngineLoader.loadSchemaEngine(
-      {
-        datamodels,
-      },
-      debug,
-      adapter,
-    )
-    return new SchemaEngineWasm({ ...rest, engine, errorRegistry: adapter.errorRegistry })
+    try {
+      const engine = await wasmSchemaEngineLoader.loadSchemaEngine(
+        {
+          datamodels,
+        },
+        debug,
+        adapter,
+      )
+      return new SchemaEngineWasm({ ...rest, engine, errorRegistry: adapter.errorRegistry })
+    } catch (e) {
+      throw adapter.errorRegistry.lookupError(e as Error) || e
```

Comment:
nit
```suggestion
      throw adapter.errorRegistry.lookupError(e as Error) ?? e
```

Reply:
Aso `throw adapter.errorRegistry.lookupError(e as Error)` is probably wrong, because `errorRegistry.lookupError` doesn't return an `Error`, it returns an `ErrorRecord`, which is defined as `{ error: unknown }`.

```suggestion
      throw adapter.errorRegistry.lookupError(e as Error)?.error ?? e
```